ksqldb-engine/src/main/java/io/confluent/ksql/internal/QueryStateListener.java

```diff
@@ -16,97 +16,132 @@
 package io.confluent.ksql.internal;
 
 import com.google.common.base.Ticker;
-import io.confluent.ksql.query.ErrorStateListener;
+import io.confluent.ksql.engine.QueryEventListener;
+import io.confluent.ksql.metastore.MetaStore;
 import io.confluent.ksql.query.QueryError;
+import io.confluent.ksql.query.QueryId;
+import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.util.QueryMetadata;
 import java.util.Collections;
 import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 import org.apache.kafka.common.MetricName;
 import org.apache.kafka.common.metrics.Gauge;
 import org.apache.kafka.common.metrics.Metrics;
 import org.apache.kafka.streams.KafkaStreams.State;
 
-public class QueryStateListener implements ErrorStateListener {
+public class QueryStateListener implements QueryEventListener {
```

Comment:
nit: could we rename to `MetricsReportingListener`?

Reply:
definitely - the diff was already too big :)